### PR TITLE
GeneralWallet supports verify_legality

### PR DESCRIPTION
**Pull request by huazhouwang**
## What does this implement/fix? Explain your changes.
GeneralWallet 支持 verify_legality

## Does this close any currently open issues?  
If it fixes a bug or resolves a feature request, be sure to link to that issue.
https://github.com/OneKeyHQ/TaskHub/issues/1904

## Pull request type
 
_Put an `x` in the boxes that apply_
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Where has this been tested?
…

## Any other comments?
…

### CODE REVIEW

File: electrum_gui/common/basic/exceptions.py [hunk 1]
```diff
@@ -51,6 +51,10 @@ class UnavailableEthAddr(OneKeyException):
     key = "msg__incorrect_eth_address"
 
 
+class UnavailableAddress(OneKeyException):
+    key = "msg__incorrect_address"
```
Comment (taimanhui) [inline electrum_gui/common/basic/exceptions.py:55]: 你们这些class和key的名字是故意想逼死强迫症啊... 😂😂😂

File: electrum_gui/android/console.py [hunk 2]
```diff
@@ -3201,7 +3202,27 @@ def verify_legality(self, data, *, flag="seed", coin="btc", password=None):  # n
...
         chain_affinity = _get_chain_affinity(coin)
-        if chain_affinity == "btc":
+        if coin and is_coin_migrated(coin):
+            chain_code = coin_manager.legacy_coin_to_chain_code(coin)
+            chain_info = coin_manager.get_chain_info(chain_code)
+            if flag == "private":
+                data = eth_utils.remove_0x_prefix(data)
+                try:
+                    data = bytes.fromhex(data)
+                    secret_manager.verify_key(chain_info.curve, prvkey=data)
```
Comment (taimanhui) [inline electrum_gui/android/console.py:3212]: 其实我们应该考虑提供`verify_hex_key`的版本然后可以直接raise对应的异常，不用全集中在入口这里做，不过这是后话了，以后看到再改过来吧

**Review by taimanhui — APPROVED** [review]

File: electrum_gui/common/basic/exceptions.py [hunk 3]
```diff
@@ -51,6 +51,10 @@ class UnavailableEthAddr(OneKeyException):
     key = "msg__incorrect_eth_address"
 
 
+class UnavailableAddress(OneKeyException):
+    key = "msg_incorrect_address"
```
[the remark this answers is not in the document]
Comment (reepig) [reply electrum_gui/common/basic/exceptions.py]: 已加。
这个 key 是用在什么地方？地址格式校验吗？

[the remark this answers is not in the document]
Comment (taimanhui) [reply electrum_gui/common/basic/exceptions.py]: 是的，地址格式校验，和上述eth/bitcoin那两个场景是一样的，但我们之前没有用统一的key是个问题。其实这里应该统一成这个`msg__incorrect_address`就可以了，不应该每个链都写一条key来表示相同的错误，因为具体对应的是哪个链的地址事实上前端应该能感知到...不过现在分离得还不清晰所以就没办法统一...

**Comment by huazhouwang** [comment]
并入 https://github.com/OneKeyHQ/electrum/pull/798

**Pull request closed without merging**